### PR TITLE
crimson/osd/osd_operations: snaptrim_event cleanup

### PR DESCRIPTION
**Pull request by Matan-B**
Cleanup (previous fix attempt) and possible regression fix 9c0432b456a69e0fbf2d9ee8135db7c0386c59b5

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

**Review by athanatos — APPROVED** [review]

File: src/crimson/osd/osd_operations/common/pg_subop_blocker.h [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+namespace crimson::osd {
+
+using interruptor =
+  ::crimson::interruptible::interruptor<
+    ::crimson::osd::IOInterruptCondition>;
+
+// bases on 998cb8c141bb89aafae298a9d5e130fbd78fe5f2
+template <typename T>
+struct SubOpBlocker : crimson::BlockerT<SubOpBlocker<T>> {
```
Comment (athanatos) [inline src/crimson/osd/osd_operations/common/pg_subop_blocker.h:17]: I don't actually see anything osd/pg specific here -- should the file be subop_blocker.h?

Comment (Matan-B) [reply src/crimson/osd/osd_operations/common/pg_subop_blocker.h]: Moved to common and renamed

**Comment by Matan-B** [comment]
https://pulpito.ceph.com/matan-2024-01-10_15:38:24-crimson-rados-wip-matanb-crimson-snaptrim_event-cleanup-distro-crimson-smithi/

**Comment by cyx1231st** [comment]
sorry about the delay, lgtm.

**Pull request merged**